### PR TITLE
added authorization

### PR DESCRIPTION
**Pull request by victordudyk**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -30,3 +27,4 @@
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+backend/src/entities/AppSettings.ts
```
Comment (barakd) [inline .gitignore:30]: shouldn't this be a json?

File: backend/src/appSettings.example.json [hunk 2]
```diff
@@ -1,4 +1,13 @@
 {
+  "authentification": {
+    "cookiesSecret": "##",
+	"cookiesMaxAge": "60000"
+    "google": {
```
Comment (barakd) [inline backend/src/appSettings.example.json]: the example json is malformed, please validate it's a valid json

File: backend/src/routes/Authentifications.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { Router } from "express";
+import AuthentificationsController from '../controllers/AuthentificationsController';
```
Comment (barakd) [inline backend/src/routes/Authentifications.ts]: typo should be AuthenticationController
without the `f` and trailing `s`

File: backend/src/services/AuthentificationService.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import AuthentificationResult from '@entities/AuthentificationResult';
```
Comment (barakd) [inline backend/src/services/AuthentificationService.ts]: same for this file name AuthenticationService.ts

File: frontend/src/App.tsx [hunk 5]
```diff
@@ -25,23 +27,27 @@ function App() {
           <Switch>
             <Route
               exact
+              component={withAuthentification(<Redirect to="/projects" />)}
```
Comment (barakd) [inline frontend/src/App.tsx]: should be withAuthentication

File: frontend/src/components/hocs/withAuthentification.tsx [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+import { getIsLoggedIn } from '../../store/duck/auth';
+
+const withAuthentification = (Component: any) => (props: any) => {
```
Comment (barakd) [inline frontend/src/components/hocs/withAuthentification.tsx]: rename to withAuthentication

**Review by barakd — APPROVED** [review]
overall looking good, approved, added a few style/typo comments, please resolve them before we merge :)

File: .gitignore (hunk 1 above)
Comment (victordudyk) [reply .gitignore:30]: fixed

File: backend/src/routes/Authentifications.ts (hunk 3 above)
Comment (victordudyk) [reply backend/src/routes/Authentifications.ts]: fixed

File: frontend/src/App.tsx (hunk 5 above)
Comment (victordudyk) [reply frontend/src/App.tsx]: fixed

File: frontend/src/components/hocs/withAuthentification.tsx (hunk 6 above)
Comment (victordudyk) [reply frontend/src/components/hocs/withAuthentification.tsx]: fixed

**Pull request merged**